homeassistant/components/enigma2/media_player.py

```diff
@@ -131,7 +132,11 @@ def state(self):
         """Return the state of the device."""
         if self.e2_box.is_recording_playback:
             return STATE_PLAYING
-        return STATE_OFF if self.e2_box.in_standby else STATE_ON
+        if self.e2_box.in_standby:
+            return STATE_OFF
+        if self.e2_box.is_offline:
+            return STATE_UNAVAILABLE
```

Comment:
Unavailable is implemented as a separate property, not as a returnable state.

Example:

```py
    @property
    def available(self):
      return self.e2_box.is_offline
````

Reply:
so, if the tuner is deep off then available property return FALSE and state property return STATE_ON ???

Reply:
The state doesn't change (well... the point is... if the device has gone offline, we simply don't know).

However, if a device is unavailable, it will be reflected in the state as well (which will become unavailable, regardless of the last state).